lib/portage/getbinpkg.py

```diff
@@ -355,26 +345,28 @@ def match_in_array(array, prefix="", suffix="", match_both=1, allow_overlap=0):
 
     for x in array:
         add_p = 0
-        if prefix and (len(x) >= len(prefix)) and (x[: len(prefix)] == prefix):
+        x_size = len(x)
+        prefix_size = len(prefix)
+        if prefix and x_size >= prefix_size and x[:prefix_size] == prefix:
             add_p = 1
 
         if match_both:
             if prefix and not add_p:  # Require both, but don't have first one.
                 continue
-        else:
-            if add_p:  # Only need one, and we have it.
-                myarray.append(x[:])
-                continue
+        elif add_p:  # Only need one, and we have it.
+            myarray.append(x[:])
+            continue
 
+        suffix_size = len(suffix)
         if not allow_overlap:  # Not allow to overlap prefix and suffix
-            if len(x) >= (len(prefix) + len(suffix)):
+            if len(x) >= (prefix_size + suffix_size):
```

Comment:
Does `x` change between here and the `x_size = len(x)` assignment above? Can we use `x_size` here?

Reply:
Yep, we can reuse `x_size` here. Coding fatigue can really mess with you! 😊